app/assets/javascripts/discourse/routes/build-category-route.js.es6

```diff
@@ -63,9 +63,15 @@ export default (filter, params) => {
 
     setupController(controller, model) {
       const topics = this.get('topics'),
-            periodId = topics.get('for_period') || (filter.indexOf('/') > 0 ? filter.split('/')[1] : '');
+            periodId = topics.get('for_period') || (filter.indexOf('/') > 0 ? filter.split('/')[1] : ''),
+            canCreateTopic = topics.get('can_create_topic'),
+            canCreateTopicOnCategory = model.get('permission') === Discourse.PermissionType.FULL;
 
-      this.controllerFor('navigation/category').set('canCreateTopic', topics.get('can_create_topic'));
+      this.controllerFor('navigation/category').setProperties({
+        canCreateTopicOnCategory: canCreateTopicOnCategory,
+        cannotCreateTopicOnCategory: !canCreateTopicOnCategory,
```

Comment:
You can add a property to `controller/navigation/category.js.es6` like this:

    cannotCreateTopicOnCategory: Ember.computed.not('canCreateTopicOnCategory'),